src/coingecko/Coingecko.ts

```diff
@@ -113,15 +115,28 @@ export class Coingecko {
 
   async call(path: string) {
     const sendRequest = async () => {
+      const { host, proHost } = this;
+
+      // Try basic, and then pro as fallback.
+      this.logger.debug({ at: "sdk-v2/coingecko", message: `Sending GET request to host ${host}` });
+      const basicUrl = `${host}/${path}`;
+
       try {
-        const { host } = this;
-        const url = `${host}/${path}`;
-        const result = await axios(url, { params: { x_cg_pro_api_key: this.apiKey } });
-        this.logger.debug({ at: "sdk-v2/coingecko", message: `Sent GET request to url ${result.request.responseURL}` });
+        const result = await axios(basicUrl, { timeout: 200 });
         return result.data;
       } catch (err) {
-        const msg = get(err, "response.data.error", get(err, "response.statusText", "Unknown Coingecko Error"));
-        throw new Error(msg);
+        this.logger.debug({
+          at: "sdk-v2/coingecko",
+          message: `Basic CG url request failed, falling back to CG PRO host ${proHost}`,
+        });
+        const proUrl = `${proHost}/${path}`;
+        try {
+          const result = await axios(proUrl, { params: { x_cg_pro_api_key: this.apiKey } });
+          return result.data;
+        } catch (errPro) {
+          const msg = get(err, "response.data.error", get(err, "response.statusText", "Unknown Coingecko Error"));
+          throw new Error(msg);
+        }
       }
     };
     return retry(sendRequest, this.numRetries, this.retryDelay);
```

Comment:
If I am reading correctly, the inbuilt retry behaviour here could result in 8 requests to CoinGecko occurring (in the worst-case scenario of all requests failing). It looks like the requests would be interleaved between free and pro.

Reply:
Yeah you're right, should we remove the retry? Or perhaps, only retry against pro api?

Reply:
Actually, it would be tricky to implement the "retry only against pro api", any suggestions?

Reply:
we could lower the retry count?

Reply:
Changed the retry logic to set retry count to 0 if apiKey is present